src/librustc_typeck/diagnostics.rs

```diff
@@ -3426,6 +3426,37 @@ parameters. You can read more about it in the API documentation:
 https://doc.rust-lang.org/std/marker/struct.PhantomData.html
 "##,
 
+E0393: r##"
+A type parameter which references `Self` in its default value was not specified.
+Example of erroneous code:
```

Comment:
I won't agree with you on this one. I think it's too much "decoration" for such a simple sentence.